hedged.go

```diff
@@ -13,37 +13,43 @@ const infiniteTimeout = 30 * 24 * time.Hour // domain specific infinite
 // NewClient returns a new http.Client which implements hedged requests pattern.
 // Given Client starts a new request after a timeout from previous request.
 // Starts no more than upto requests.
-func NewClient(timeout time.Duration, upto int, client *http.Client) *http.Client {
+// cancelContext controls whether or not this client will cancel request contexts automatically.
+//   It is recommened to set this to true unless you are having issues.
+func NewClient(timeout time.Duration, upto int, cancelContext bool, client *http.Client) *http.Client {
```

Comment:
I think it will be much easier for the final user to use function without this param, so we will have `NewClient` as it was and `NewClientNoContext` (better name?). WDYT?

Reply:
Same applies to the `NewRoundTripper`.